rust/extractor/src/config.rs

```diff
@@ -34,23 +36,39 @@ pub struct Config {
     pub scratch_dir: PathBuf,
     pub trap_dir: PathBuf,
     pub source_archive_dir: PathBuf,
+    pub log_dir: PathBuf,
     pub extract_dependencies: bool,
     pub verbose: u8,
     pub compression: Compression,
     pub inputs: Vec<PathBuf>,
+    pub qltest: bool,
+    pub qltest_cargo_check: bool,
 }
 
 impl Config {
     pub fn extract() -> anyhow::Result<Config> {
         let args = argfile::expand_args(argfile::parse_fromfile, argfile::PREFIX)
             .context("expanding parameter files")?;
         let cli_args = CliConfig::parse_from(args);
-        Figment::new()
+        let mut figment = Figment::new()
             .merge(Env::prefixed("CODEQL_"))
             .merge(Env::prefixed("CODEQL_EXTRACTOR_RUST_"))
             .merge(Env::prefixed("CODEQL_EXTRACTOR_RUST_OPTION_"))
-            .merge(Serialized::defaults(cli_args))
-            .extract()
-            .context("loading configuration")
+            .merge(Serialized::defaults(cli_args));
+        if matches!(figment.find_value("qltest"), Ok(Value::Bool(_, true))) {
```

Comment:
Wouldn't `if let Ok(Value::Bool(_, true)) = figment.find_value("qltest")` do the same thing?

Reply:
true, let me try that